File: module-domain/src/main/java/eu/geekhome/domain/configurable/DoubleFieldBuilder.java
package eu.geekhome.domain.configurable;

public class DoubleFieldBuilder implements FieldBuilder<Double> {

    @Override
    public Double fromPersistableString(String value) {
        if (value == null) {
            return null;
        }

        try {
            return Double.parseDouble(value);
        } catch (NumberFormatException nfe) {
            return null;
        }
    }

    @Override
    public String toPersistableString(Double value) {
        return value.toString();
    }
}

File: module-domain/src/main/java/eu/geekhome/domain/configurable/DoubleFieldBuilder.kt
package eu.geekhome.domain.configurable

import java.lang.NumberFormatException

class DoubleFieldBuilder : FieldBuilder<Double> {
    override fun fromPersistableString(value: String?): Double {
        return if (value == null) {
            0.0
        } else try {
            value.toDouble()
        } catch (nfe: NumberFormatException) {
            0.0
        }
    }

    override fun toPersistableString(value: Double): String {
        return value.toString()
    }
}